microservice/src/test/java/de/gsi/microservice/concepts/majordomo/MajordomoWorker.java

```diff
@@ -0,0 +1,299 @@
+        case W_HEARTBEAT:
+            // Do nothing for heartbeats
+            break;
+        case W_DISCONNECT:
+            reconnectToBroker();
+            break;
+        case W_UNKNOWN:
+        default:
+            // N.B. not too verbose logging since we do not want that sloppy clients can bring down the broker through warning or info messages
+            if (LOGGER.isDebugEnabled()) {
+                LOGGER.atDebug().addArgument(uniqueID).addArgument(request).log("worer '{}' received invalid message: '{}'");
+            }
+            break;
+        }
+        return null;
+    }
+
+    public boolean isVerbose() {
+        return verbose;
+    }
+
+    public MdpMessage processRequest(final MdpMessage request, final byte[] clientSourceID) {
+        if (requestHandler != null) {
+            // de-serialise
+            // byte[] -> PropertyMap() (+ getObject(Class<?>))
+            try {
+                final byte[][] payload = requestHandler.handle(request.payload);
+                // serialise
+                return new MdpWorkerMessage(request.senderID, MdpWorkerCommand.W_REPLY, serviceBytes, clientSourceID, payload);
+            } catch (Exception e) { // NOPMD on purpose since we want to catch exceptions and courteously return this to the user
+                final StringWriter sw = new StringWriter();
+                final PrintWriter pw = new PrintWriter(sw);
+                e.printStackTrace(pw);
+
+                final StringBuilder builder = new StringBuilder();
+                builder.append(MajordomoWorker.this.getClass().getName())
+                        .append(" caught exception in user-provided call-back function for service '")
+                        .append(getServiceName())
+                        .append("'\nrequest msg: ")
+                        .append(request)
+                        .append("\nexception: ")
+                        .append(sw.toString());
+                final String exceptionMsg = builder.toString();
+                return new MdpWorkerMessage(request.senderID, MdpWorkerCommand.W_REPLY, serviceBytes, clientSourceID, exceptionMsg.getBytes(StandardCharsets.UTF_8));
+            }
+        }
+        return null;
+    }
+
+    public void registerHandler(final RequestHandler handler) {
+        this.requestHandler = handler;
+    }
+
+    @Override
+    public void run() {
+        while (!Thread.currentThread().isInterrupted()) {
+            handleReceive();
+        }
+        destroy();
+    }
+
+    public void setReconnect(int reconnect) {
+        this.reconnect = reconnect;
+    }
+
+    public void setVerbose(final boolean verbose) {
+        this.verbose = verbose;
+    }
+
+    @Override
+    public synchronized void start() {
+        reconnectToBroker();
+        super.start();
+    }
+
+    /**
+     * Send reply, if any, to broker and wait for next request.
+     */
+    protected void handleReceive() {
```

Comment:
Codacy found an issue: [The method 'handleReceive()' has an NPath complexity of 554, current threshold is 200](https://app.codacy.com/manual/GSI/chart-fx/pullRequest?prid=6307742)